### PR TITLE
Created news as described by #17

### PR DESCRIPTION
**Pull request by rfvcft**
The biggest thing is how I chose to do patch-updates, following the answer here: https://stackoverflow.com/questions/63143731/update-sqlalchemy-orm-existing-model-from-posted-pydantic-model-in-fastapi

The gist of it is:

```
for var, value in vars(data).items():
  setattr(news, var, value) if value else None
```

meaning to go through the variable-value pairs of the updated news, and change the non-null attributes.

### CODE REVIEW

File: services/news_service.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+def update_existing_news(news_id: int, data: NewsUpdate, db: Session):
+    data.pinned_from, data.pinned_to = validate_pinned_times(data.pinned_from, data.pinned_to)
+
+    news = db.query(News_DB).filter_by(id=news_id).one_or_none()
+    if news is None:
+        raise HTTPException(status.HTTP_404_NOT_FOUND)
+
+    # This does not allow one to "unset" values that could be null but aren't currently
+    for var, value in vars(data).items():
```
Comment (PelicanQ) [inline services/news_service.py:52]: Okay, but it is a nice solution to doing many if's on all update schema fields. We will have to see if we ever feel like unsetting.

**Comment by PelicanQ** [comment]
I just realized that just saying `str | None` on an incoming schema will not set omitted JSON fields to None. We also gotta do `some_field: str | None = None` to make it actually optional to include in the body.
See this one: https://docs.pydantic.dev/latest/migration/#required-optional-and-nullable-fields

Also, I added `default=None`  to `author_id` so that you can actually create a News_DB(...) without an author_id. Otherwise the constructor will require it.

**Review by PelicanQ — APPROVED** [review]

**Comment by rfvcft** [comment]
I actually wanted author_id to not be allowed to be None. I was thinking that a News should always have an author. In the news_create route that is always set to the current user, and in regular use cases that should be the only way to create a new News. In the other cases, I believe it is better to force an author.

**Comment by PelicanQ** [comment]
Oh okay. To require a field to not be null don't use `Mapped[Optional[<some type>]]`
I made it required here: 
https://github.com/fsek/WebWebWeb/commit/8e173cf9f9e1dc1c576c54e6a417bb47fbceda60
Also note that if you want the user object in a route **and** require specific permission, use Permission.require(...) as a function argument.

**Pull request merged**